src/main/java/org/noureddine/joularjx/monitor/MonitoringHandler.java

```diff
@@ -59,6 +64,14 @@ public void run() {
                 var methodsStats = extractStats(samples, methodName -> true);
                 var methodsStatsFiltered = extractStats(samples, properties::filtersMethod);
 
+                //Collecting call trees stats only if the option is enabled
+                Map<Thread, Map<CallTree, Integer>> callTreesStats = null;

```

Comment:
Map object creation should only be done if option is enable. Put them inside the condition

Reply:
The map creation is outside the condition because if it is inside, there will be a compilation failure because of the usage of those maps further at lines 99, 100, 104 and 105.